core/client/src/main/java/alluxio/client/file/FileOutStream.java

```diff
@@ -103,8 +103,13 @@ public FileOutStream(AlluxioURI path, OutStreamOptions options, FileSystemContex
       mUnderStorageOutputStream = null;
     } else { // Write is through to the under storage, create mUnderStorageOutputStream
       try {
-        WorkerNetAddress worker = // not storing data to Alluxio, so block size is 0
-            options.getLocationPolicy().getWorkerForNextBlock(mBlockStore.getWorkerInfoList(), 0);
+        WorkerNetAddress worker;
+
+        while ((worker = // not storing data to Alluxio, so block size is 0
```

Comment:
Why does this have to be in a loop, and could this loop forever? (not enough workers)